### ISSUE TITLE
Uncaught [RequestError code: INVALID_PARAMETER, message: Invalid parameter 'file'. Must exist.]

### ISSUE THREAD

**Issue by pulyaevskiy** [dart-atom/dartlang#679]
This happened to me a couple times, but I'm not sure I have solid steps to reproduce. There is only a few things I can tell:

1. The issue is when I try to execute a test file with `Cmd+R` of `F5`, I get the error below.
2. Started after latest upgrade (0.5.1)
3. This seems to happen with multiple projects opened though I'm not sure if it's relevant
4. Restarting Atom and opening just one project helps and the error disappears for a while.
5. Sometimes I can get error with `Cmd+R`, but `F5` works fine.

Hope this helps.

**Atom Version**: 1.3.2
**System**: Mac OS X 10.11.2
**Thrown From**: [dartlang](https://github.com/dart-atom/dartlang) package, v0.5.1

### Stack Trace

Uncaught [RequestError code: INVALID_PARAMETER, message: Invalid parameter 'file'. Must exist.]

```
At /Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:22548

[RequestError code: INVALID_PARAMETER, message: Invalid parameter 'file'. Must exist.]
    at [object Object].dart.wrapException (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:23597:17)
    at _rootHandleUncaughtError_closure.dart._rootHandleUncaughtError_closure.call$0 (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:44566:19)
    at [object Object].dart._microtaskLoop (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:42177:27)
    at [object Object].dart._startMicrotaskLoop (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:42183:11)
    at TimerImpl_internalCallback0.dart.TimerImpl_internalCallback0.call$0 (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:22912:23)
    at invokeClosure_closure.dart.invokeClosure_closure.call$0 (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:24765:29)
    at _IsolateContext.dart._IsolateContext.eval$1 (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:22539:25)
    at [object Object].dart._callInIsolate (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:22183:28)
    at dart.invokeClosure (/Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:23745:18)
    at /Users/anatoly/.atom/packages/dartlang/web/entry.dart.js:23766:18
```

### Commands

```
     -0:24.9.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:24.9.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:21.6.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:21.6.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:21.4.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:21.4.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:15.8.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -0:14.3.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:14.3.0 dartlang:pub-get (atom-text-editor.editor)
     -0:06.8.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:06.8.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:05.8.0 dartlang:run-application (atom-text-editor.editor.is-focused)
     -0:04.5.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:04.5.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:03.4.0 dartlang:debug-run (atom-text-editor.editor.is-focused)
     -0:00.9.0 dartlang:run-application (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "language-javascript-jsx"
    ],
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ],
    "followSymlinks": false
  },
  "dartlang": {
    "sdkLocation": "/usr/local/opt/dart/libexec",
    "showTodos": true,
    "formatOnSave": true
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v0.2.10
atom-material-ui, v0.8.0
dartlang, v0.5.1
file-icons, v1.6.13
flutter, v0.1.1
git-log, v0.4.1
git-tab-status, v1.9.2
language-api-blueprint, v1.0.2
language-docker, v1.1.6
linter, v1.11.3
linter-api-blueprint, v0.2.0
pretty-json, v0.4.1
project-manager, v2.7.1
react, v0.12.10
synced-sidebar, v0.4.3
travis-ci-status, v0.18.0

# Dev
No dev packages
```

**Comment by devoncarew** [comment]
Looks like that stack trace unfortunately doesn't have a lot of info. Do you have any breakpoints set? The error message - `RequestError code: INVALID_PARAMETER` - _might_ be something sent by the observatory.

**Comment by pulyaevskiy** [comment]
I don't think I had any breakpoints set when it occurred last time. 

I've just tried to test breakpoints and the issue didn't surface. However there is one oddity, which is seemingly related to the `mockito` package.

In one of my test files I have mocks. If don't set any breakpoints execution works as expected. But if I set at least one then:

1. Execution doesn't stop on that breakpoint
2. Process finishes (it seems) and debug window closes.
3. Debug log window (in the bottom pane) stays
4. For each run I get plus one more debug log window in the bottom pane and number of processes also increases by one (in the status bar).

Not sure if this is related again. Will be observing this further and will let you know if I find something.
